### PR TITLE
[B2BORG-76] Modify orgs functionality for adding vs editing user accounts

### PR DESCRIPTION
**Pull request by arturmagalhaesjr**
# B2BORG-76

> Currently there is a single mutation for adding & editing B2B users. This is causing some unexpected issues.

The purpose of this PR is to add those 2 mutations:
- Add user; 
- Update user;

Also, the structure of this app was reorganized and some unnecessary variables were removed;

Each one of them, calls its respective storefront permissions method.

Those mutations have not been used on front-end yet. Therefore you have to test it by using the playground:

**Select the organizations-graphql app**

GraphQL Playground: [https://b2borg--sandboxusdev.myvtex.com/admin/graphql-ide](https://b2borg--sandboxusdev.myvtex.com/admin/graphql-ide)

#### Mutation
```graphql
mutation ($name:String! $email:String! $orgId:ID! $costId:ID! $roleId:ID! $canImpersonate:Boolean) {
  addUser(    
    roleId: $roleId,    
    orgId: $orgId,
    costId: $costId    
    canImpersonate:$canImpersonate
    name: $name
    email:$email
  ) {
    id
  }
}
```
#### Variables

``` json
{
  "canImpersonate": false,
  "name": "Foo Bar zz",
  "email": "foo@bar.zz",
  "orgId": "e25541b8-b67a-11ec-835d-0a47005fd913",
  "costId": "e53bc3fd-b67a-11ec-835d-0ad48d7ed21d",
  "roleId": "f2a38526-f3bd-11eb-82ac-12fda6a1f41f"
}
```

#### Mutation

```graphql
mutation ($clId: ID! $userId:ID $orgId:ID! $costId:ID! $roleId:ID! $canImpersonate:Boolean $id:ID!) {
  updateUser(    
    id:$id,
    clId: $clId,
    roleId: $roleId,    
    orgId: $orgId,
    costId: $costId    
    canImpersonate:$canImpersonate
    userId: $userId
  ) {
    id
  }
}
``` 
#### Variables
``` json
{
  "canImpersonate": false,
  "id": "6aadf0a8-b819-11ec-835d-02e42b7c762d",
  "roleId": "f4f6dd62-f3bd-11eb-82ac-02bea657840d",
  "userId": null,
  "clId": "69d001c0-b819-11ec-835d-02918d36d5d3",
  "orgId": "e25541b8-b67a-11ec-835d-0a47005fd913",
  "costId": "e53bc3fd-b67a-11ec-835d-0ad48d7ed21d"
}
```

### CODE REVIEW

File: node/clients/storefrontPermissions.ts [hunk 1]
```diff
@@ -64,13 +64,26 @@ export const QUERIES = {
 }
 
 export const MUTATIONS = {
+  addUser: `mutation ($id: ID $userId: ID $roleId: ID! $orgId: ID! $costId: ID! $name: String! $canImpersonate: Boolean! $email: String!) {
+    addUser (id:$id, userId: $userId, roleId: $roleId, orgId: $orgId, costId: $costId, name: $name, canImpersonate: $canImpersonate, email: $email) {
+      id
```
Comment (polishq) [inline node/clients/storefrontPermissions.ts:69]:
```suggestion
      id
      status
      message
```

File: node/clients/storefrontPermissions.ts [hunk 2]
```diff
@@ -64,13 +64,26 @@ export const QUERIES = {
...
     saveUser(id: $id, roleId: $roleId, userId: $userId, orgId: $orgId, costId: $costId, clId: $clId, canImpersonate: $canImpersonate, name: $name, email: $email) {
       id
       status
       message
     }
   }`,
+
+  updateUser: `mutation ($id: ID $clId: ID! $userId: ID $roleId: ID! $orgId: ID! $costId: ID! $canImpersonate: Boolean!) {
+    updateUser (id:$id, clId: $clId, userId: $userId, roleId: $roleId, orgId: $orgId, costId: $costId, canImpersonate: $canImpersonate) {
+      id
```
Comment (polishq) [inline node/clients/storefrontPermissions.ts:83]:
```suggestion
      id
      status
      message
```

**Review by polishq — CHANGES_REQUESTED** [review]
In addition to the two minor changes in the comments, I think we missed something when we updated `vtex.storefront-permissions`. Part of the reason for splitting the `saveUser` mutation into `add` and `update` mutations was so that we could prevent a user from being moved from one organization to another (either accidentally or maliciously). With the latest code I'm still able to "add" a user that already belongs to another organization or has a different role in the same organization, and their existing org/cost/role will be overwritten with the new ones. The "add" operation should only be allowed if the user has no org/cost/role assigned.

**Review by arturmagalhaesjr — COMMENTED** [review]
Done!
Also, I've opened this PR: https://github.com/vtex-apps/storefront-permissions/pull/43/files to cover the feature that I mentioned regarding a validation.

**Review by polishq — APPROVED** [review]
LGTM

**Pull request merged**